src/app/tests/TestThreadBorderRouterManagementCluster.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/CommandHandler.h>
+#include <app/clusters/thread-border-router-management-server/thread-border-router-management-server.h>
+#include <cstdint>
+#include <cstring>
+#include <lib/core/CHIPError.h>
+#include <lib/support/BitFlags.h>
+#include <lib/support/Span.h>
+#include <lib/support/ThreadOperationalDataset.h>
+#include <lib/support/tests/ExtraPwTestMacros.h>
+#include <pw_unit_test/framework.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace GeneralCommissioning {
+// Mock function
+void SetBreadcrumb(Attributes::Breadcrumb::TypeInfo::Type breadcrumb) {}
+} // namespace GeneralCommissioning
+
+namespace ThreadBorderRouterManagement {
+// Mock functions
```

Comment:
These are not mock --> they are globally overridden by having the test target depend only on the single CPP file again, instead of correctly separating the dependencies.

The delegate interface should should have a `bool IsFailSafeArmed()`, which would allow the platform to connect it properly. This would remove the need to:
- Split IsFailSafeArmed in a different module
- Having a test target that directly depends on the cluster source, rather than the test having a `deps` on the cluster business logic itself

Reply:
I make the server instance include a Failsafe context reference so that we will not need the two functions